File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0;
        int buy=prices[0];
        for(auto &price:prices){
            if(price-buy>=0){
                profit=max(profit,price-buy);
            }
            else {
                buy=price;
            }
        }
        return profit;
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit=0
        n=len(prices)
        buy=float('inf')
        for price in prices:
            if price-buy>=0:
                maxprofit=max(maxprofit,price-buy)
            else:
                buy=price
            
        return maxprofit